### ISSUE TITLE
Synchronisation between weboverview of PodFetch and AntennaPod does not work

### ISSUE THREAD

**Issue by update-freak** [SamTV12345/PodFetch#384]
### Describe the bug

I listened to a podcast in the web overview of the podfetch-instance and the podcast episode is marked as viewed in the overview but not in the AntennaPod-App.
I created an gpodder-Account with the command line with /app/podfetch users add.
In AntennaPod I'm logged in to this gpodder-Account and created the device name "Podcasts".
I'm wondering how I can select this device name "Podcasts" in the web overview of the podfetch-instance -> maybe that's the reason why the podcast episode is not marked as read in the AntennaPod-App?

![windows](https://github.com/SamTV12345/PodFetch/assets/40196995/5d62d90b-80d2-4987-8049-d15e4dadbccb)
![antennapod](https://github.com/SamTV12345/PodFetch/assets/40196995/06e04321-bf1e-4122-96fc-1bfc1b1e2247)

When I use the synchronisation between two Android-devices with AntennaPod and the gPodder-Account with the same device "Podcasts" the finished-listened podcasts are synchronised.

### Reproduction

1. create a gpodder-Account in Podfetch 2. Login to this account in AntennaPod 3. Add the podcasts from AntennaPod to PodFetch. 4. Finish a episode in PodFetch 5. Check if the episode is also finished in AntennaPod

### System Info

```Shell
Chrome: 119
AntennaPod: 3.1.2
PodFetch: 4.2.2
System: DSM 7.2.1 on Docker on Synology NAS
```

### Used Package Manager

n/a

### Validations

- [X] Follow our [Code of Conduct](https://github.com/SamTV12345/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/SamTV12345/.github/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by SamTV12345** [comment]
@update-freak Could you try the latest image and then run upon start `podfetch migration episodes`. This should hopefully resolve the issue with webview and app and vice versa.

**Comment by update-freak** [comment]
yes, watchtower updated to the latest-image and I tried with `podfetch migration episodes`. Then I restarted the container PodFetch and PodFetch-DB. Then I did a complete synchronisation on AntennaPod, but the podcast is in AntennaPod still not marked as finished.

**Comment by SamTV12345** [comment]
@update-freak Can you access the database and show the content of the episode table?

**Comment by update-freak** [comment]
How I can access to the database. Which tools are necessary?

**Comment by SamTV12345** [comment]
You can add this section to your docker-compose:

```
 pgadmin:
    image: dpage/pgadmin4
    container_name: pgadmin4_container
    restart: always
    ports:
      - "8888:80"
    environment:
      PGADMIN_DEFAULT_EMAIL: user-name@domain-name.com
      PGADMIN_DEFAULT_PASSWORD: strong-password
    volumes:
      - pgadmin-data:/var/lib/pgadmin
  ```

and access the portal at http://<your-ip:8888 

There you need to add the postgres database with the credentials and then you can open tables and check the 'episodes' table

**Comment by update-freak** [comment]
I can not connect to the DB. I'm not sure what I did wrong. Maybe you see the mistake.

![2023-11-07 20_33_58-Portainer _ local](https://github.com/SamTV12345/PodFetch/assets/40196995/145c8bd9-5251-427d-a5e7-934e49d7c5f1)
![2023-11-07 20_33_38-pgAdmin 4](https://github.com/SamTV12345/PodFetch/assets/40196995/6f7f57da-57e3-4d3d-ab17-95988d7b57f0)
![2023-11-07 20_33_19-Portainer _ local](https://github.com/SamTV12345/PodFetch/assets/40196995/8e56de3e-8bb8-4f55-9452-65f79916324f)

**Comment by SamTV12345** [comment]
Try to change the hostname to postgres and the port to 5432. 

The problem with the hostname is that postgres doesn't communicate via http, so you only need the hostname, in your case the ip address of the docker machine. The other thing is the port number. Based on the docker-compose screenshot you haven't exposed the port 4135 to the host machine.

**Comment by update-freak** [comment]
tried with this config (also with hostname = postgres) but same result:

![2023-11-07 22_32_07-pgAdmin 4](https://github.com/SamTV12345/PodFetch/assets/40196995/a28c51ea-8b7f-41ce-b6af-a06b9bd0d6b0)

Full docker-compose is this one:
`
version: '3.9'
services:
  podfetch:
    image: samuel19982/podfetch:postgres
    container_name: PodFetch
    user: 1026:100
    ports:
      - 4135:8000
    volumes:
      - /volume1/docker/podfetch:/app/podcasts:rw
    environment:
      - POLLING_INTERVAL=300
      - SERVER_URL=https://podfetch.beispieladresse.dynv6.net
      - DATABASE_URL=postgresql://postgres:changeme@postgres/podfetch
      - BASIC_AUTH=true
      - USERNAME=Felix
      - PASSWORD=z1Pj!aszpy#fLe8n
      - GPODDER_INTEGRATION_ENABLED=true
    restart: always

  postgres:
    image: postgres
    container_name: PodFetch-DB
    security_opt:
      - no-new-privileges:true
    healthcheck:
      test: ["CMD", "pg_isready", "-q", "-d", "podfetch", "-U", "postgres"]
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: changeme
      POSTGRES_DB: podfetch
    volumes:
      - /volume1/docker/podfetch/db:/var/lib/postgresql/data:rw
    restart: always
`

**Comment by SamTV12345** [comment]
I'll try it now.

**Comment by SamTV12345** [comment]
This was the config I meant:

<img width="705" alt="image" src="https://github.com/SamTV12345/PodFetch/assets/40429738/bc6655b2-de9d-4e50-9983-e3652b7b7ded">

**Comment by update-freak** [comment]
Any news on that. Please let me know if I can help

**Comment by SamTV12345** [comment]
> Any news on that. Please let me know if I can help

If you could send me a screenshot of the episodes table that would be dope.

**Comment by update-freak** [comment]
Tried it again, also with your proposal with several times with "postgres" but both cases did not work. I will try to ask in different forums to get an answer about that.
![2023-11-12 14_55_01-pgAdmin 4](https://github.com/SamTV12345/PodFetch/assets/40196995/70284cf6-3039-4387-ba03-b02623abe211)

**Comment by SamTV12345** [comment]
Weird. If everything is inside that docker-compose.yaml, it should work. I just tried it again. With hostname postgres it just works.


```yaml
version: '3.9'
services:
  podfetch:
    image: samuel19982/podfetch
    container_name: PodFetch
    user: 1026:100
    ports:
      - 4135:8000
    volumes:
      - /volume1/docker/podfetch:/app/podcasts:rw
    environment:
      - POLLING_INTERVAL=300
      - SERVER_URL=https://podfetch.beispieladresse.dynv6.net
      - DATABASE_URL=postgresql://postgres:changeme@postgres/podfetch
      - BASIC_AUTH=true
      - USERNAME=Felix
      - PASSWORD=z1Pj!aszpy#fLe8n
      - GPODDER_INTEGRATION_ENABLED=true
    restart: always

  postgres:
    image: postgres
    container_name: PodFetch-DB
    security_opt:
      - no-new-privileges:true
    healthcheck:
      test: ["CMD", "pg_isready", "-q", "-d", "podfetch", "-U", "postgres"]
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: changeme
      POSTGRES_DB: podfetch
    volumes:
      - /volume1/docker/podfetch/db:/var/lib/postgresql/data:rw
    restart: always
  pgadmin:
    image: dpage/pgadmin4
    container_name: pgadmin4_container
    restart: always
    ports:
      - "8888:80"
    environment:
      PGADMIN_DEFAULT_EMAIL: user-name@domain-name.com
      PGADMIN_DEFAULT_PASSWORD: strong-password
    volumes:
      - pgadmin-data:/var/lib/pgadmin
 ```

**Comment by update-freak** [comment]
now it worked. Problem was that the pgadmin4-Code was not at the same stack. I created first an extra stack, therefore I could not find the DB. The question is how to get the "episodes table" when I connected to the DB? :D

**Comment by SamTV12345** [comment]
> now it worked. Problem was that the pgadmin4-Code was not at the same stack. I created first an extra stack, therefore I could not find the DB. The question is how to get the "episodes table" when I connected to the DB? :D
> 
> EDIT: found it. ![2023-11-12 17_45_36-Einstellungen](https://user-images.githubusercontent.com/40196995/282313049-14286d1d-f1fa-4f68-8fdf-619fd64f704c.png) ![2023-11-12 17_47_07-Podfetch](https://user-images.githubusercontent.com/40196995/282313050-098a2926-1cb1-4023-b9f0-3908c08ba1be.png) ![Screenshot_20231112_174647_AntennaPod](https://user-images.githubusercontent.com/40196995/282313052-30578495-b3ab-4156-a59d-9e9157f47909.jpg)

Nice great that this works. I need the output of cleaned_urls.

**Comment by update-freak** [comment]
Here is the podcast_episodes.csv. If there is a smarter way to get the cleaned_urls let me know.
[podcast_episodes.csv](https://github.com/SamTV12345/PodFetch/files/13329045/podcast_episodes.csv)

**Comment by SamTV12345** [comment]
Ok. There are a lot of episodes. I am wondering what caused it to save text in the cleaned_url column. Try runnning `/app/podfetch migration episode` multiple times. No idea why it seems to abort at some point.

**Comment by update-freak** [comment]
I tried it and got this screen, but I'm not sure if this is sucessfully ended because I did not get any feedback. I attached the csv-file which I downloaded again. 

![2023-11-13 18_44_46-NAS - Synology DiskStation](https://github.com/SamTV12345/PodFetch/assets/40196995/773c5229-6d8f-4cb2-a69f-f5e3fe025293)

[podcast_episodes.csv](https://github.com/SamTV12345/PodFetch/files/13339317/podcast_episodes.csv)

When this will not work, I can also try to make a clean install again.

**Comment by SamTV12345** [comment]
Thanks for your efforts helping me fix this issue. I think you sent maybe the wrong table. It should be episodes.  The cleaned_url column is the thing I am most interested as it is a smaller or equal representation of the podcast episode link. Meaning they should always be linked together.

**Comment by update-freak** [comment]
No problem. Thank you also for the great tool and your effort.
Jop, I found the correct one I guess. Here is it:
[episodes.csv](https://github.com/SamTV12345/PodFetch/files/13341766/episodes.csv)

**Comment by update-freak** [comment]
Do you need any further input?

**Comment by SamTV12345** [comment]
No. Thanks for asking 😄. I got an assignment for my studies which I had to prepare. I'll have a look at it tomorrow.

**Comment by SamTV12345** [comment]
> Do you need any further input?

Good question why it is not working. The urls are correctly set. Can you send me the sqldump via email samelus1998@outlook.de

To do that you login into pgadmin

1. Click on the podfetch table via rightclick -> Backup
2. In the window enter a filename, e.g. podfetch.sql, select format plain and click on "Backup"
3. Click on the top navbar on Tools-> Storage Manager, select the file and click the Download icon at the top.

**Comment by update-freak** [comment]
thanks for the steps. I send the sql-dump to you.

**Comment by SamTV12345** [comment]
Thanks for the sql dump. I'll test it out today. From a quick first look they should link together.

**Comment by SamTV12345** [comment]
Do you also listen to the podcasts in the webview with your user instead of the admin user account?

**Comment by update-freak** [comment]
is this assumption corret?
admin account: initally created with docker compose (username= ..., passwort=...) ----> Felix
user account: /app/podfetch users add                                                                     ----> felix

I used the admin account (Felix) to add the podcasts via opml-file.
In AntennaPod I used the user account (felix) to login
--> maybe the problem is the same name (Felix, felix) or the problem could be that I have to take the account felix also to check the listened-status.

Because I thought that it would be better to use an other name for the created one with " /app/podfetch users add" I added an podcasts Account, but here the login does not work, even when the role is admin:

![2023-11-21 07_42_31-Podfetch](https://github.com/SamTV12345/PodFetch/assets/40196995/c44dd71f-f314-4e9e-8d65-3035223e9d4c)
![2023-11-21 07_42_45-Einstellungen](https://github.com/SamTV12345/PodFetch/assets/40196995/8d2ac6e5-d499-4cee-b13e-3401b1d3e3f4)
